[Feature] Add elastic client

GROwen opened the pull request:
### Motivation
Need to clean-up stale indices on the Elastic Cloud development deployment.

### Changes
1. Adds the `elastic-cloud` command and `delete-stale` sub-command

GROwen commented on cmd/elastic-cloud/index.go:
```diff
@@ -0,0 +1,92 @@
+		for k, i := range list {
+			if strings.Contains(k, "elasticsearch_index") {
+				now := time.Now().UnixMilli()
+				created, err := strconv.ParseInt(i.IndexItem.IndexDetail.CreationDate, 10, 64)
```
Using the creation date is a very rudimentary way of identifying stale indices because it does not identify the last time an index was updated.

However because content is ephemeral and these are development indices we don't need to be too concerned about destructive requests.

nicksantamaria commented on README.md:
```diff
@@ -21,6 +21,24 @@ This will store the encrypted file at `keys/production/oauth.pem.asc`.
+> Variables are deployment specific - make sure the deployment you are targeting is not a production deployment.
+
+* EC_API_KEY - Generated from the deployments Kibana settings
+* EC_CLOUD_ID - Found on the deployments Elastic Cloud 'manage' page
```
Would this be better as a flag or argument instead of an envvar?

GROwen replied:
Because they're creds I thought it safer to keep them stored as env vars but I suppose we could set them as env vars and reference them in the argument. Am I being too cautious in not allowing a user to expose those vars directly in an arg on the cli (and therefore making it available in their shell history)?

GROwen commented:
@nicksantamaria I've refactored the inputs so the API_KEY is now an env var and Cloud ID a flag. I've also added the age of the indices to target as flag.

Ready for review again.

nicksantamaria commented:
@GROwen see https://github.com/dpc-sdp/bay-cli/pull/11
